packages/fx-core/src/plugins/resource/apim/factory.ts

```diff
@@ -168,11 +168,16 @@ export class Factory {
       "credential",
       await azureAccountProvider?.getAccountCredentialAsync()
     );
-    const apiManagementClient = new ApiManagementClient(credential, solutionConfig.subscriptionId);
+    const subscriptionInfo = await azureAccountProvider?.getSelectedSubscription();
+    AssertNotEmpty("subscriptionInfo", subscriptionInfo);
```

Comment:
Let `subscriptionInfo=AssertNotEmpty("subscriptionInfo", subscriptionInfo);`
so `subscriptionInfo.subscriptionId` can be used without `!`

Reply:
Good idea, but ts compile check does not recognize ```AssertNotEmpty```.